test/provider/database.unittest.ts

```diff
@@ -0,0 +1,222 @@
+import 'jest';
+import 'fake-indexeddb/auto';
+
+import {Database, CollectionMap} from '../../src/provider/model/database/Database';
+import {StoredNotification} from '../../src/provider/model/StoredNotification';
+import {NotificationInternal} from '../../src/client/internal';
+import {Collection} from '../../src/provider/model/database/Collection';
+
+let database: Database;
+let collection: Collection<StoredNotification>;
+let generatedNotificationCount: number = 0;
+
+beforeAll(async () => {
+    database = await new Database().delayedInit();
+    collection = database.get(CollectionMap.NOTIFICATIONS);
+});
+
+describe('Database Methods', () => {
+    describe('Get', () => {
```

Comment:
Minor point, but can these `describe`s explain the behaviour, rather than just grouping tests?

e.g:
`"Get"` => `"When fetching"`
`"Upsert"` => `"When updating"`, etc.

Ideally, the `describe` and `it` names should read as a sentence when combined.